flax/optim/base.py

```diff
@@ -534,7 +532,10 @@ def __init__(self, filter_fn):
 
   @staticmethod
   def _get_params_dict(inputs):
-    if isinstance(inputs, base.Model):
+    if inputs.__class__.__name__ == 'Model':
```

Comment:
Wouldn't it be easier to move the deprecation warning to some piece of the old Module code that is always triggered when you actually use it? Like the Module constructor? Other codebases might also important the old nn api for simple instance checks and such or other backward compatibility tasks.

Reply:
Good point. That's much simpler indeed. /cc @levskaya.